WordPress/Classes/ViewRelated/Post/Scheduling/CalendarCollectionView.swift

```diff
@@ -0,0 +1,134 @@
+import Foundation
+import JTAppleCalendar
+
+class CalendarCollectionView: JTACMonthView {
+
+    let calDataSource = CalendarDataSource()
+
+    override init() {
+        super.init()
+
+        register(DateCell.self, forCellWithReuseIdentifier: "dateCell")
+
+        backgroundColor = .clear
+
+        scrollDirection = .horizontal
+        scrollingMode = .stopAtEachCalendarFrame
+        showsHorizontalScrollIndicator = false
+
+        ibCalendarDataSource = calDataSource
+        ibCalendarDelegate = calDataSource
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+
+        register(DateCell.self, forCellWithReuseIdentifier: "dateCell")
+
+        scrollDirection = .horizontal
+        scrollingMode = .stopAtEachCalendarFrame
+        showsHorizontalScrollIndicator = false
+
+        ibCalendarDataSource = calDataSource
+        ibCalendarDelegate = calDataSource
+    }
+}
+
+class CalendarDataSource: JTACMonthViewDataSource {
+
+    var willScroll: ((DateSegmentInfo) -> Void)?
+    var didScroll: ((DateSegmentInfo) -> Void)?
+    var didSelect: ((Date) -> Void)?
+
+    func configureCalendar(_ calendar: JTACMonthView) -> ConfigurationParameters {
+        let startDate = Date.farPastDate
+        let endDate = Date.farFutureDate
+        return ConfigurationParameters(startDate: startDate, endDate: endDate)
+    }
+}
+
+extension CalendarDataSource: JTACMonthViewDelegate {
+    func calendar(_ calendar: JTACMonthView, cellForItemAt date: Date, cellState: CellState, indexPath: IndexPath) -> JTACDayCell {
+        let cell = calendar.dequeueReusableJTAppleCell(withReuseIdentifier: "dateCell", for: indexPath) as! DateCell
+        configureCell(view: cell, cellState: cellState)
+        return cell
+    }
+
+    func calendar(_ calendar: JTACMonthView, willDisplay cell: JTACDayCell, forItemAt date: Date, cellState: CellState, indexPath: IndexPath) {
+        configureCell(view: cell, cellState: cellState)
+    }
+
+    func calendar(_ calendar: JTACMonthView, willScrollToDateSegmentWith visibleDates: DateSegmentInfo) {
+        willScroll?(visibleDates)
+    }
+
+    func calendar(_ calendar: JTACMonthView, didScrollToDateSegmentWith visibleDates: DateSegmentInfo) {
+        didScroll?(visibleDates)
+    }
+
+    func calendar(_ calendar: JTACMonthView, didSelectDate date: Date, cell: JTACDayCell?, cellState: CellState, indexPath: IndexPath) {
+        configureCell(view: cell, cellState: cellState)
+        didSelect?(date)
+    }
+
+    func calendar(_ calendar: JTACMonthView, didDeselectDate date: Date, cell: JTACDayCell?, cellState: CellState, indexPath: IndexPath) {
+        configureCell(view: cell, cellState: cellState)
+    }
+
+    private func configureCell(view: JTACDayCell?, cellState: CellState) {
+       guard let cell = view as? DateCell else { return }
```

Comment:
Closing braces should always be placed on a new line regardless of the number of enclosed statements